Pylint can freeze when running in the background and encounters
an 'import readline' statement.

The reason is that pylint is trying to import everything it encounters, hence
invoking import init functions. Python readline does call GNU readline init,
which is supposed to be called only when terminal is available (and hence forces
the SIGTTOU signal).

We can safely ignore this signal, because readline is not used further.

See bug 31397460 for more details.

--- pylint-2.6.0/pylint/__init__.py
+++ pylint-2.6.0/pylint/__init__.py
@@ -18,6 +18,9 @@ from pylint.__pkginfo__ import version a
 def run_pylint():
     from pylint.lint import Run as PylintRun
 
+    import signal
+    signal.signal(signal.SIGTTOU, signal.SIG_IGN)
+
     try:
         PylintRun(sys.argv[1:])
     except KeyboardInterrupt:
